### DISCUSSION TITLE
Decode QRCode in .NET6

### DISCUSSION THREAD

**Discussion by nayankariya** [micjahn/ZXing.Net#456] [Q&A] [answered]
Hello Everyone,

I'm trying to implement decoding of QRCode in my .NET6 application using ZXing.Net and have below questions:

1) Is ZXing.Net is supported for .NET6? I was having my application in .NET 4.5 version and everything was working great. Now, on .NET 6, I am not able to use same code. 

Here is my code: 

string[] files = Directory.GetFiles("wwwroot/qrr");
foreach (string file in files)
{
    // create a barcode reader instance
    IBarcodeReader reader = new BarcodeReader();
    BarcodeReader reader = new BarcodeReader();
    // load a bitmap
    var barcodeBitmap = (Bitmap)Image.FromFile("wwwroot/qrr/" + Path.GetFileName(file));

    // detect and decode the barcode inside the bitmap
    var result = reader.Decode(barcodeBitmap);
}

I have downloaded latest Nuget package for .NET6 and Decode method is not working anymore. Can someone please help me out?

Thanks in advance.

**Comment by Woetoo** [comment]
I've had a similar problem (except the opposite way around - where I'm encoding/writing).
[Unable to code "New BarcodeWriter" in VB.NET (.Net 6) with v0.16.8](https://github.com/micjahn/ZXing.Net/discussions/449)

If I'm representing the answer correctly, the .NET packages are split out somewhat. With the core of ZXing being in the package you've probably already downloaded, but with things like the bitmap handlers in their own packages for different configurations or operating systems.

Try incorporating:
https://www.nuget.org/packages/ZXing.Net.Bindings.Windows.Compatibility

Though their are other choices if you search "ZXing.Net.Bindings" within NuGet (for example, if you're using Android).

I'm very new at this. So decoding may be a separate solution.

**Comment by micjahn** [comment, marked as answer]
@Woetoo is right.
The .Net Framework until version 4.8 contains the implementation of System.Drawing.Bitmap. The default implementations of the BarcodeReader and BarcodeWriter classes use that type. So your sample works only with the main ZXing.Net package until version 4.8 of the .Net framework.
.Net Standard, .Net Core and .Net 5.0 and above change significantly. The core framework doesn't include a Bitmap class. 
So the ZXing.Net base package for the framework versions doesn't contain a reader or writer class for Bitmaps. 
The implementations for specific image manipulation libraries are now in separate packages (ZXing.Net.Bindings.*).
You have to select the package which fit your needs. If you want to use ImageSharp for processing image data you add the package ZXing.Net.Bindings.ImageSharp to your project. If you want to use SkiaSharp, you have to add https://www.nuget.org/packages/ZXing.Net.Bindings.SkiaSharp.
And if you want to use System.Drawing.Common, you have to add https://www.nuget.org/packages/ZXing.Net.Bindings.Windows.Compatibility.
And so on. The missing classes reside in their specific namespaces like ZXing.Windows.Compatibility, ZXing.ImageSharp or ZXing.SkiaSharp.

**Reply by nayankariya** [reply]
Thank you - that helped me with my issue. Appreciate it.
